fortuna/src/keeper.rs

```diff
@@ -0,0 +1,348 @@
+use {
+    crate::{
+        api,
+        chain::{
+            ethereum::SignablePythContract,
+            reader::{
+                BlockNumber,
+                EntropyReader,
+                RequestedWithCallbackEvent,
+            },
+        },
+        config::EthereumConfig,
+    },
+    anyhow::Result,
+    ethers::{
+        middleware::NonceManagerMiddleware,
+        providers::{
+            Http,
+            Middleware,
+            Provider,
+            StreamExt,
+        },
+        types::U256,
+    },
+    std::sync::Arc,
+    tokio::{
+        sync::mpsc,
+        time::{
+            self,
+            sleep,
+            Duration,
+        },
+    },
+    tracing,
+};
+
+pub async fn process_event(
+    event: RequestedWithCallbackEvent,
+    chain_config: &api::BlockchainState,
+    contract: &Arc<SignablePythContract>,
+    nonce_manager: &Arc<NonceManagerMiddleware<Provider<Http>>>,
+    gas_limit: U256,
+) -> Result<()> {
+    let provider_revelation = chain_config.state.reveal(event.sequence_number)?;
+
+    let sim_res = chain_config
+        .contract
+        .simulate_reveal(
+            event.provider_address,
+            event.sequence_number,
+            event.user_random_number,
+            provider_revelation,
+        )
+        .await;
+
+    match sim_res {
+        Ok(_) => {
+            let res = contract
+                .reveal_with_callback_wrapper(
+                    event.provider_address,
+                    event.sequence_number,
+                    event.user_random_number,
+                    provider_revelation,
+                    nonce_manager.next(),
```

Comment:
we should use nonce manager to send these transactions instead of manually getting the next nonce